### ISSUE TITLE
[BUG] InventoryItemActionEvent is not registering id

### ISSUE THREAD

**Issue by Krulvis** [powbot/issues#153]
**Describe the bug**

`InventoryItemActionEvent` is supposed to register the id of the clicked item but it always returns 0.

**To Reproduce/Line of code causing issue**
Add the following to a test script, run it and click an item in your inventory. It will return id=0.
```
@com.google.common.eventbus.Subscribe
    fun onGameActionEvent(e: GameActionEvent) {
        log.info("$e")
    }
```

**Expected behavior**
InventoryItemActionEvent.id is whatever the id of the item you clicked on is.

**Logs**
```
[INFO] InventoryItemActionEvent(name=Knife, id=0, interaction=Use)
[INFO] InventoryItemActionEvent(name=Knife -> Maple longbow (u), id=0, interaction=Use)
```

**Comment by constt** [comment]
@Krulvis Had a look and just seems that's the way the client works now. Is using names an option?

**Comment by Krulvis** [comment]
Yes, I suggest removing the ID field from InventoryItemActionEvent and make InventoryItemActionEvent.item() return an item using the name then. Only downside is that you won't be able to distinguish between noted/unnoted :(

**Comment by constt** [comment]
@Krulvis removing that field would break a lot of scripts from compiling.

What if the item is consumed by the event, item would be nil?

**Comment by Krulvis** [comment]
Yes, that would be best

**Comment by Krulvis** [comment]
Here is the code for using name. you have to take into account that after using one item the 'name' of the second InventoryItemActionEvent becomes firstItem -> secondItem:
```kotlin
fun InventoryItemActionEvent.item(): Item? {
    val itemName = if (name.contains("->")) name.substring(name.indexOf("-> ") + 3) else name
    return Inventory.stream().name(itemName).firstOrNull()
}
```
